### DISCUSSION TITLE
Custom server configuration for self-hosted linux Ubuntu

### DISCUSSION THREAD

**Discussion by eldieco** [prisma/prisma#2276] [Q&A]
Hello Prisma Community!

I have a huge favor to ask. I hope someone much smarter than me can help out. And please forgive my ignorance. I'm a front-end developer by trade, so when it comes to server configuration, I'm lost in the woods. My strength is in UX design and writing code, not fiddling with servers.

Ordinarily, when I deploy an app, I use a 3rd party service like Heroku or Now (Vercel). This time around, I'm having to help set up a self-hosted prisma environment because the sys admin/account manager wants to use his own server and he's never heard of prisma. 

All I have access to is the remote server (where the app is supposed to live) via SSH and a linux command line. No desktop or OS GUI, nothing....Anyone have any clue how to get a prisma deployment working in such a scenario or would you be willing to assist?

We need to get this up and running to demo for the client, and none of the documentation I've read is helping.

**Here are some additional caveats:**

The way I currently have to access the server is by SSHing from my local computer (a Mac) into a remote machine and then SSH into another server with a local IP (192.168.....) running Ubuntu. This is where the app is supposed to be deployed, but I can't seem to even get `prisma deploy` to work with Docker on this machine. How can I even test if a web app is working if I can't view it in a web browser via a publicly-facing IP or on a desktop environment with a web browser? The sys admin who owns the server doesn't seem to understand that. Is he missing something or am I?

**I'd be happy to take this discussion offline** if someone feels like they could really offer some help here. Thanks a bunch!!

Eldie.

**Comment by ryands17** [comment]
Hey @eldieco :wave: 
Is this project hosted somewhere on GitHub?
If so, could you clone this on the remote machine and then install all the dependencies via `npm` or `yarn` and then run the app?

**Comment by eldieco** [comment]
Hey Ryan! Yes it is on Github. And yes I am able to get that far by cloning
it to the server and npm install and all that. The issue I'm running into
is the server's IP is not publicly accessible (does not ping). So I can't
seem to get the Prisma docker container to connect to the localhost.  I
can't seem to convey to the sys admin that I would need a working IP
address in order to deploy a web application. Am I missing something here
or am I correct in saying that one would need to be able to view a WEB site
inside a WEB browser in order to test if eveything is working properly?
He's got me SSHing into one machine and proxied into another with a local
IP (192.168). It's all a little over my head. I was able to get the demo
deployed to Heroku in the meantime, and it's working great. But I'm not
sure he's going to want to use Heroku because he's being difficult.

**Comment by ryands17** [comment]
If you want to deploy the application on the server for others to view and use, then it has to have a public IP. Without that you wouldn't be able to test it outside of that server.

You can still test the application while ssh'ed in the server though. You just need to start the application and it would run locally on the specified port that you provide it and you could check it that way. But it would only be accessible in that server, not outside.

For e.g. While using GraphQL Yoga with Prisma2, on starting the server, you would get a running playground where you can test your queries/mutations. But that would only be available in the browser if that machine has a public IP address, otherwise it wouldn't be available for those not ssh'ed into the server.

Let me know if you have any queries!

**Comment by eldieco** [comment]
To be honest, I'm not even sure how to access it from within the server
while SSH'd because it's only a terminal interface. So how can i bring up
the playground on that server if I can only view the server from a command
line interface? That's where I'm getting confused here. Very little of
actual development for this project happens via the command line, but now
I'm supposed to do this all through a terminal? I'm confused lol. There's
not gnome desktop set up on this server or anything and so I can only
interact via command line which is a little outside my wheelhouse for this
project.

**Comment by ryands17** [comment]
To test it via the command line, the ony option would be via `curl` sending a simple GraphQL request while ssh'ed in the terminal. And if it works, you can inform your sysadmin to open the server publicly so that you can view it from your browser.

Something like this

```sh
curl \
  -X POST \
  -H "Content-Type: application/json" \
  --data '{ "query": "{ posts { title } }" }' \
  http://localhost:4000/graphql
```

Where the PORT would be something that you've set
